packages/openapi3-parser/lib/parser.js

```diff
@@ -126,14 +126,30 @@ function parse(source, context) {
     R.unless(isObjectOrAnnotation, createError(context.namespace, 'Source document is not an object')),
     R.unless(isAnnotation, parseOpenAPIObject(context)));
 
-  return R.chain(
+  const parseResult = R.chain(
     R.pipe(
       parseDocument,
       deduplicateUnsupportedAnnotations(context.namespace),
       context.options.generateSourceMap ? filterColumnLine : filterSourceMaps
     ),
     document
   );
+
+  if (!isAnnotation(parseResult.content[0])) {
+    const formatVersion = (/(?<=(["']?)openapi\1\s*:\s*(["']?))(3\.\d+\.\d+)/).exec(source)[0];
```

Comment:
This regex still have false positives, I do not think it is a good idea to use regex for this and I would recommend follow my suggestion from before in the prior review (https://github.com/apiaryio/api-elements.js/pull/533#pullrequestreview-466602157) by using the parsed YAML/JSON output object instead of raw source. It will be more reliable.

This regex is not counting for all possible cases (note that the source can be either YAML or JSON). With some valid YAML inputs I am seeing this logic throw an error:

```yaml
"openapi": '3.0.0'
info:
  title: hello
  version: 3.2.0
paths: {}
```

```
TypeError: Cannot read property '0' of null
    at parse (/Users/kyle/Projects/apiaryio/api-elements.js/packages/openapi3-parser/lib/parser.js:139:92)
    at /Users/kyle/Projects/apiaryio/api-elements.js/packages/openapi3-parser/lib/adapter.js:29:21
    at new Promise (<anonymous>)
    at Object.parse (/Users/kyle/Projects/apiaryio/api-elements.js/packages/openapi3-parser/lib/adapter.js:25:10)
    at Fury.parse (/Users/kyle/Projects/apiaryio/api-elements.js/packages/core/lib/fury.js:230:29)
    at FuryCLI.run (/Users/kyle/Projects/apiaryio/api-elements.js/packages/cli/lib/fury.js:104:23)
    at Object.<anonymous> (/Users/kyle/Projects/apiaryio/api-elements.js/packages/cli/lib/fury.js:243:11)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
```

There is other cases for false positives:

Input:

```yaml
info:
  title: hello
  version: 3.2.0
paths:
  /parse:
    post:
      requestBody:
        content:
          application/openapi+json:
            example:
              openapi: 3.0.0
      responses:
        '204':
          description: result
openapi: 3.0.1
```

Incorrectly returns openapi 3.0.0 instead of 3.0.1:

```json
{
  "element": "link",
  "meta": {
    "title": {
      "element": "string",
      "content": "OpenAPI 3.0.0"
    }
  },
  "attributes": {
    "relation": {
      "element": "string",
      "content": "via"
    },
    "href": {
      "element": "string",
      "content": "https://spec.openapis.org/oas/v3.0.0"
    }
  }
}
```

Reply:
ok, I misunderstood

Reply:
changed https://github.com/apiaryio/api-elements.js/pull/533/commits/48abc84718745440b7d268d55ccb291a5467f16f